llvm/test/Feature/Repo/repo_pruning_dependents.ll

```diff
@@ -1,30 +1,51 @@
+; during the pruning.
+;
+; The steps in this test are:
+; Step 1: Remove the database.
+; Step 2: Generate an IR code targeted on Repo.
+; Step 3: Remove the TicketNode metadata from the functions B and C.
+; Step 4: Since the functions B and C don't have the TicketNode, the digest of functions `B`
+;         and `C` are generated by the RepoObjectWriter and added these two functions into
+;         the dependents of the function 'A'.
+; Step 5: Dump all compilation members in the database. There are 3 compilation members,
+;         which are functions `A`, `B` and `C`.
+; Step 6: Re-build the same source IR code again. In this step, the IR code firstly pass the
+;         RepoMetadataGenerator pass to generate the digests of functions 'A', 'B' and 'C'.
+;         During the RepoPruning pass, the function 'A' is pruned and its dependents 'B' and
+;         'C' are added.
+; Step 7: Dump all compilation members in the database again. The compilations, which are
+;         generated by step 4 and step 6, should be the same. Therefore, there are only one
+;         compilation, which includes three members: functions `A`, `B` and `C`.
+; Step 8: Check that the step 5 and step 7 generate the same compilation members.
 ;
-; If function 'A' is dependent on function 'B' and 'B' is dependent
-; on function 'C', both TickeNodes of 'B' and 'C' need to be added
-; into in the 'repo.tickets' during the pruning.
-
-; The testcase includes three steps:
-; Step 1: Build the code and create the database 'clang.db' which contains all Tickets.
-; Step 2: Dump all compilation members in the database;
-; Step 3: Re-build the same IR code again.
-; Step 4: Dump all compilation members in the database again;
-; Step 5: Check that the step 2 and step 4 generate the same compilation members.
-
-; This test only works for the Debug build because the digest of A is calculated for the Debug build.
-
 ; RUN: rm -f %t.db
-; RUN: env REPOFILE=%t.db llc -mtriple="x86_64-pc-linux-gnu-repo" -filetype=obj %s -o %t
-; RUN: env REPOFILE=%t.db pstore-dump  --all-compilations %t.db > %t.log
-; RUN: env REPOFILE=%t.db clang -O3 -c --target=x86_64-pc-linux-gnu-repo -x ir %s  -o %t1
-; RUN: env REPOFILE=%t.db pstore-dump  --all-compilations %t.db > %t1.log
-; RUN: diff %t.log %t1.log
-
-; REQUIRES: asserts
+; RUN: env REPOFILE=%t.db clang -O0 -S -emit-llvm --target=x86_64-pc-linux-gnu-repo -x ir %s  -o %t.repo.ll
+; RUN: %python %S/Inputs/remove_GO_ticketnode.py -f B C -i %t.repo.ll -o %t.ll
+; RUN: env REPOFILE=%t.db llc -mtriple="x86_64-pc-linux-gnu-repo" -filetype=obj %t.ll -o %t.1.o
+; RUN: env REPOFILE=%t.db pstore-dump  --all-compilations %t.db > %t.1.log
+; RUN: env REPOFILE=%t.db clang -O0 -c --target=x86_64-pc-linux-gnu-repo -x ir %s -o %t.2.o
+; RUN: env REPOFILE=%t.db pstore-dump  --all-compilations %t.db > %t.2.log
+; RUN: diff %t.1.log %t.2.log
 
-target triple = "x86_64-pc-linux-gnu-repo"
+target triple = "x86_64-pc-linux-gnu-elf"
```

Comment:
Again, why the change to target ELF? We explicitly ask for repo in the lines above.

Reply:
The same reason as mentioned above.